app/src/main/java/ie/macinnes/tvheadend/player/HtspDataSource.java

```diff
@@ -229,6 +229,7 @@ public int read(byte[] buffer, int offset, int readLength) throws IOException {
                     Log.v(TAG, "Blocking for more data ("+mDataSourceNumber+")");
                 Thread.sleep(250);
             } catch (InterruptedException e) {
+                //TODO: Clean up and re-throw, or java will do it anway
```

Comment:
There's nothing to clean up or rethrow here, when we're InterruptedException, it's expected we just stop waiting for data - and return 0 (as we read no data into the buffer)